UnityProject/Assets/Scripts/HealthV2/Living/Surgery/SurgeryProcedureBaseSingleton.cs

```diff
@@ -3,11 +3,48 @@
 using ScriptableObjects;
 using UnityEngine;
 
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
 namespace HealthV2
 {
 	[CreateAssetMenu(fileName = "SurgeryProcedureBaseSingleton", menuName = "ScriptableObjects/Surgery/SurgeryProcedureBaseSingleton")]
 	public class SurgeryProcedureBaseSingleton : SingletonScriptableObject<SurgeryProcedureBaseSingleton>
 	{
 		public List<SurgeryProcedureBase> StoredReferences = new List<SurgeryProcedureBase>();
+
+#if UNITY_EDITOR
+		[NaughtyAttributes.Button()]
+		public void FindAll()
+		{
+
+			StoredReferences.Clear();
+			EditorUtility.SetDirty(this);
+			var SurgeryProcedureBases = FindAssetsByType<SurgeryProcedureBase>();
+			StoredReferences.AddRange(SurgeryProcedureBases);
+			EditorUtility.SetDirty(this);
+			Undo.RecordObject(this, "AlertSOs FindAll ");
+		}
+
+		public static List<T> FindAssetsByType<T>() where T : UnityEngine.Object
```

Comment:
I think this can be more useful if it's put elsewhere in our utilities scripts instead of here.

Reply:
as editor namespace so it's a pain to integrate